File: test-suite/src/test/java/io/micronaut/docs/aop/lifecycle/ProductBean.java
package io.micronaut.docs.aop.lifecycle;

// tag::imports[]
import io.micronaut.aop.*;
import io.micronaut.context.annotation.Prototype;
import java.lang.annotation.*;
// end::imports[]

// tag::class[]
@Retention(RetentionPolicy.RUNTIME)
@AroundConstruct // <1>
@InterceptorBinding(kind = InterceptorKind.POST_CONSTRUCT) // <2>
@InterceptorBinding(kind = InterceptorKind.PRE_DESTROY) // <3>
@Prototype // <4>
public @interface ProductBean {
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/aop/lifecycle/ProductBean.py
# tag::imports[]
import java

from micronaut.aop import AroundConstruct, InterceptorBinding
from micronaut.context.annotation import Prototype
# end::imports[]

InterceptorKind = java.type("io.micronaut.aop.InterceptorKind")


# tag::class[]
@AroundConstruct  # <1>
@InterceptorBinding(kind=InterceptorKind.POST_CONSTRUCT)  # <2>
@InterceptorBinding(kind=InterceptorKind.PRE_DESTROY)  # <3>
@Prototype  # <4>
def ProductBean(func):
    return func
# end::class[]
